### ISSUE TITLE
Toolbox DeprecationWarning: object of type <class 'numpy.float64'> cannot be safely interpreted as an integer

### ISSUE THREAD

**Issue by balarsen** [spacepy/spacepy#181]
`toolbox.linspace()` produces a DeprecationWarning

Fix is likely to make the number input an int. Likely also an issue for `logspace`.

### Minimal example to reproduce issue:
```
In [1]: import spacepy.toolbox as tb
In [2]: tb.linspace(1, 4, 3.0)
/Users/balarsen/git/spacepy/spacepy/toolbox/__init__.py:1310: DeprecationWarning: object of type <class 'float'> cannot be safely interpreted as an integer.
  return np.linspace(min, max, num, **kwargs)
Out[2]: array([1. , 2.5, 4. ])
```

### Error message/Traceback:
```
/Users/balarsen/git/spacepy/spacepy/toolbox/__init__.py:1310: DeprecationWarning: object of type <class 'float'> cannot be safely interpreted as an integer.
  return np.linspace(min, max, num, **kwargs)
```

### OS, Python version, and dependency version information:

```
Darwin-18.6.0-x86_64-i386-64bit
sys.version_info(major=3, minor=7, micro=3, releaselevel='final', serial=0)
numpy=1.16.4
scipy=1.3.0
matplotlib=3.1.0
```

**Comment by drsteve** [comment]
So this is a `numpy` warning that is exposed by us wrapping `numpy.linspace` to provide a little extra funcationality. We, like `numpy`, give the expected type in the docs. Hiding a cast to integer is probably okay, especially since that's basically what happens in numpy if you call it with a float number of steps right now, but I don't see this as a bug or warning that's in our court.

So I'll mull over #182 a little before merging. The question is whether we should have a hidden cast-to-int or whether we should expect the user to follow the documented types (and then live with the change in `numpy` behaviour when they stop allowing this). We _could_ (and I don't actually think I like this idea) do the cast if there's no remainder modulo 1, otherwise raise an exception.

**Comment by balarsen** [comment]
Seems to me our goal should be user friendly and ease. Making the cast
appropriate. But discuss on.

**Comment by drsteve** [comment]
Let's say someone is calling this in a function, or something, and their code ends up passing floats to this argument by mistake. Imagine the floats are the wrong variable, and have non-integer values as well as non-integer types. If we cast, that will continue to silently pass and potentially not do what the users expected.

Now, there's a clear argument for retaining current behaviour. But there's also, IMO, a pretty good argument for raising an exception if they supply a type that will potentially give unexpected results (which is where `numpy` is moving its behaviour to). Personally, I don't think I've ever hit a case where I ended up passing a non-int to that argument.

Maybe the thing to do is, if it's not an int, then cast to int and raise a warning. That way if users are interested they can either fix or catch the warnings.

**Comment by balarsen** [comment]
So I think you are right and instead of the cast in linspace/logspace where it would hide I hunted down the place that linspace/logspace were called with float inside spectrogram and fixed it there which is better.

**Comment by drsteve** [comment]
Fixed in #182
